src/NuGet.Core/NuGet.Packaging/Signing/Signatures/PrimarySignature.cs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+#endif
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public abstract class PrimarySignature : Signature
+    {
+#if IS_DESKTOP
+        /// <summary>
+        /// A SignedCms object holding the signature and SignerInfo.
+        /// </summary>
+        public SignedCms SignedCms { get; }
```

Comment:
This can be moved into `Signature`, since all signatures need to atleast be a `SignedCms`

Reply:
Countersignatures are represented as `SignerInfo`, so they don't have their own `SignedCms` only Primary signatures will have it, that's why I choose to leave it in this class. If we see the need for the primary signature's SignedCms in the countersignature we should consider moving it, but in the mean time I think it should stay here.